### PR TITLE
config, callbacks: add RECT and LINE tools

### PR DESCRIPTION
**Pull request by pascal-niklaus**
as requested [here](https://github.com/bk138/gromit-mpx/pull/192#issuecomment-1963718075) : 

PR implementing LINE and RECT, but not the arrowtype option.

### CODE REVIEW

File: src/main.h [hunk 1]
```diff
@@ -144,6 +146,8 @@ typedef struct
   cairo_surface_t *undobuffer[GROMIT_MAX_UNDO];
   gint            undo_head, undo_depth, redo_depth;
 
+  cairo_surface_t *temp_buffer;
```
Comment (bk138) [inline src/main.h:149]: Maybe a more speaking name or a comment above?

**Review by bk138 — COMMENTED** [review]
Thank you very very much for this fine PR! Only one minor comment.

File: src/main.h (hunk 1 above)
Comment (bk138) [reply src/main.h:149]: To me it seems it's the back buffer of the back buffer, but this might not be a good name...

**Comment by xSeekTruthx** [comment]
I don't believe this is as described by the OP, which I would love to see if possible?

- Straight mode - Click for start, Click for end. OR
- Click "Creates a dot", hold [Shift] for straight mode, Click ends/creates line.

Straight Line Overlay

**Pull request merged**